src/wp-includes/kses.php

```diff
@@ -931,19 +932,22 @@ function wp_kses_version() {
  * @global array[]|string $pass_allowed_html      An array of allowed HTML elements and attributes,
  *                                                or a context name such as 'post'.
  * @global string[]       $pass_allowed_protocols Array of allowed URL protocols.
+ * @global bool           $pass_allowed_comments  Whether HTML comments are allowed or not.
  *
  * @param string         $string            Content to filter.
  * @param array[]|string $allowed_html      An array of allowed HTML elements and attributes,
  *                                          or a context name such as 'post'. See wp_kses_allowed_html()
  *                                          for the list of accepted context names.
  * @param string[]       $allowed_protocols Array of allowed URL protocols.
+ * @param bool           $allowed_comments  Whether HTML comments are allowed or not.
  * @return string Content with fixed HTML tags
  */
-function wp_kses_split( $string, $allowed_html, $allowed_protocols ) {
-	global $pass_allowed_html, $pass_allowed_protocols;
+function wp_kses_split( $string, $allowed_html, $allowed_protocols, $allowed_comments ) {
+	global $pass_allowed_html, $pass_allowed_protocols, $pass_allowed_comments;
 
 	$pass_allowed_html      = $allowed_html;
 	$pass_allowed_protocols = $allowed_protocols;
+	$pass_allowed_comments  = $allowed_comments;
 
 	return preg_replace_callback( '%(<!--.*?(-->|$))|(<[^>]*(>|$)|>)%', '_wp_kses_split_callback', $string );
```

Comment:
given that we're setting this global, and it looks like the goal is have a convenient way of sharing state with the regex callbacks, should we store and restore its value so we don't leave the rest of the system in a dangling or partial state?

```php
global …, $pass_allowed_comments;

$prev_allowed_comments = $pass_allowed_comments;
$pass_allowed_comments = $allowed_comments;

$split = preg_replace_callback( … );

$pass_allowed_comments = $prev_allowed_comments;
```